### PR TITLE
Windows: Update pytest with echo and remove others

### PR DESCRIPTION
**Pull request by nicolecheetham**
Updating tests with echo pytest mark and removing other pytests from various test files.

### CODE REVIEW

File: lib/spack/spack/test/test_suite.py [hunk 1]
```diff
@@ -307,6 +307,7 @@ def test_test_part_fail(tmpdir, install_mockery, mock_fetch, mock_test_stage):
         assert status == TestStatus.FAILED
 
 
+@pytest.mark.not_on_windows("echo not available on Windows")
```
Comment (johnwparent) [inline lib/spack/spack/test/test_suite.py]: Echo in the sense that you can run `which` on it is not available as it is an internal command to `cmd`, I think we can still reasonably find something that will echo to stdout, either using Python or composing our use of echo to be a batch script.

**Review by johnwparent — CHANGES_REQUESTED** [review]

**Review by johnwparent — DISMISSED** [review]
I think we _could_ make the one skipped test here work,  but it's been marked for removal in the near future and could potentially be a bit tricky to make it work cross platform as echo is a shell command on Win. 

This looks good otherwise, lgtm.

**Comment by wdconinc** [comment]
Flagging merge conflict. Otherwise looks good to go.

**Comment by johnwparent** [comment]
@wdconinc merge conflict has been resolved. This looks good to go.

**Review by johnwparent — APPROVED** [review]

File: lib/spack/spack/test/test_suite.py (hunk 1 above)
Comment (wdconinc) [reply lib/spack/spack/test/test_suite.py]: @johnwparent Do you want to mark this as resolved, or do you have a suggestion here?

Comment (johnwparent) [reply lib/spack/spack/test/test_suite.py]: I've resolved it, this issue is more pervasive than just this one instance and the solution is outside the scope of this PR and should/will be done in a follow up. 

Thanks for pointing out this is unresolved @wdconinc!

**Pull request merged**